biblib/views/operations_view.py

```diff
@@ -197,60 +197,64 @@ def post(self, library):
-            if primary.public: primary_is_public = True
-            if self.read_access(service_uid=user_editing_uid,
-                                        library_id=library_uuid):
-                has_read_access_primary = True 
-        
-            for lib in data.get('libraries', []):
+            primary_is_public = primary.public
+            has_read_access_primary = self.read_access(
+                service_uid=user_editing_uid, library_id=library_uuid
+            )
+
+            secondary_libraries = data.get("libraries", [])
+            for lib in secondary_libraries:
                 try:
                     secondary_uuid = self.helper_slug_to_uuid(lib)
                 except TypeError:
                     return err(BAD_LIBRARY_ID_ERROR)
+
                 secondary = session.query(Library).filter_by(id=secondary_uuid).one()
-                if secondary.public: secondary_is_public = True
                 lib_names.append(secondary.name)
-                if self.read_access(service_uid=user_editing_uid,
-                                    library_id=secondary_uuid):
-                    has_read_access_secondary = True  
-                if self.write_access(service_uid=user_editing_uid,
-                            library_id=secondary_uuid): 
-                    has_write_access_secondary = True 
-                    
-                    
-            if data['action'] in ['union', 'intersection', 'difference']:
-                if 'libraries' not in data:
-                    return err(NO_LIBRARY_SPECIFIED_ERROR)
-                if not (secondary_is_public or has_read_access_secondary) or not has_read_access_primary: 
-                    return err(NO_PERMISSION_ERROR)
-                if 'name' not in data:
-                    data['name'] = 'Untitled {0}.'.format(get_date().isoformat())
-                if 'public' not in data:
-                    data['public'] = False
-
-            if data['action'] == 'copy':
-                
-                if 'libraries' not in data:
-                    return err(NO_LIBRARY_SPECIFIED_ERROR)
-                if len(data['libraries']) > 1:
-                    return err(TOO_MANY_LIBRARIES_SPECIFIED_ERROR)
-                # Check the permissions of the user
-                if not (primary_is_public or has_read_access_primary) or not has_write_access_secondary:
-                    return err(NO_PERMISSION_ERROR)           
-
-        
+
+                secondary_is_public = secondary.public
+
+                has_read_access_secondary = self.read_access(
+                    service_uid=user_editing_uid, library_id=secondary_uuid
+                )
+
+                has_write_access_secondary = self.write_access(
+                    service_uid=user_editing_uid, library_id=secondary_uuid
+                )
+
+                if action in ["union", "intersection", "difference"]:
+                    if not has_read_access_primary or not (
```

Comment:
Looks good! The only comment I have, and I might be misunderstanding this, is that we can also perform `union`, `intersection`, and `difference` actions with a public primary library. In that case, you might be able to move the logical check for the primary out of the loop and immediately exit if the primary permissions aren't set and then have checks only on the secondary in the loop. Would potentially save some `postgres` queries.

Reply:
For the `union`, `intersection`, and `difference` we need a public `secondary` library, so I think it makes sense as it is right now. Feel free to correct me.

Reply:
I think you are entirely correct on that point. I'm wondering if `has_read_access_primary` should be `(primary_is_public or has_read_access_primary)` as it is in the check for `copy`. If that is the case we may be able to keep the checks in the loop to only looking at the secondary because the requirements for the primary are the same.

Reply:
That's the thing. For `copy` we need `(primary_is_public or has_read_access_primary)` but for `union, intersection, difference` we need `has_read_access_primary`. 
At least that is my understanding. 

Here's my notes: 
`copy`:  

- primary: has_read_access or is_public 
- secondary: has_write_access 

`union, intersection, difference`:

- primary: has_read_access 
- secondary: has_read_access or is_public

Does that make sense to you? 
I talked to Kelly and checked a couple of times, but we might have missed something.